File: project1/collatz_conjecture.py
print("Welcome, this is collatz conjecture or simply 3x+1 calculator:")
def collatz_conjecture(num):
    sequence = [num]
    while num >1:
        if ((num % 2) == 0):
            num = num // 2
        else:
            num = num *3 +1
        sequence.append(num)
    return sequence
    
num = int(input("Enter any natural number: "))
if num>1:
    print("Sequence for", num, "is:")
    print(collatz_conjecture(num))
else:
    print(num, "is not natural number.")

File: extra projects/Html, Css, Js/JS algorithms/collatz-conjecture.js
console.log("Hello, this is collatz conjecture or simply 3x+1 calculator")
const readline = require("readline")

const collatzConjecture={
    calculation: function(userNum){
        userNum=parseInt(userNum)
        let originalUserNum=userNum;
        let sequence=[];
        let counter=1;
        while (userNum > 1){
            if (userNum%2==0){
                userNum=Math.floor(userNum / 2)
            } else{
                userNum = userNum * 3 + 1
            }
            sequence.push(`\n${counter} : ${userNum}`)
            counter++;
        }
        console.log(`Collatz conjecture sequence for ${originalUserNum} is \n ${sequence}`)
        console.log(`Total steps needed: ${counter-1}`)
    }
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

rl.question("Enter natural number: ", (userNum) => {
    if (userNum>1){
        collatzConjecture.calculation(userNum);
    } else{
        console.log(`${userNum} is not a natural number!`)
    }
    rl.close()
})